cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
It isn't very clear to me how having a label solves the problem. If the label contains the image that was originally deployed (without a digest because say pinning failed), then later when the stack file is updated (without `--no-resolve-image` flag, but the same image), how do we decide to contact the registry?

To me, the `--image` flag in `service create` and `service update` also conveys "user intent", something that I'm not able to grasp in stack files. Please correct me if I've misunderstood.